text/0039-service-overhaul.md

```diff
@@ -0,0 +1,462 @@
+    - *`"protocol name"`*<sup>O</sup>: (e.g. `"HLS":`) array of strings with codecs' name supported by only this protocol. This array is merged with `"any_protocol"`.
+  - `"audio"`: audio codecs supported by the service.
+    - `"any_protocol"`<sup>O</sup>: array of strings with codecs' name supported by the service for any protocol.
+    - *`"protocol name"`*<sup>O</sup>: (e.g. `"HLS":`) array of strings with codecs' name only supported by this protocol. This array is merged with `"any_protocol"`.
+- `"supported_resolutions"`<sup>O</sup>: array of strings with resolutions (e.g. `"1280x720"`) supported by the service for any protocol.
+  - `"with_framerates"`<sup>O</sup>: default to false. If false each entry shall be put as this "1920x1080", if true it will be put with a framerate as this "1920x1080@30". Only those framerate will be be considered supported.
+  - `"array"`(required): Array of resolutions
+- `"maximums"`<sup>O</sup>: maximums allowed by the service.
+  - `"fps"`<sup>O</sup>: maximum framerate allowed by the service. May be overrided if `"supported_resolutions"` is set with framerates (if 1080p is limited to 30 FPS and 720p to 60, the maximum will change depending on the resolution).
+  - `"video_bitrate"`<sup>O</sup>: maximum video bitrate. Can be set per codec. Unused if `"video_bitrate_matrix"` is set.
+    - `"any_codec"`<sup>O</sup>: bitrate for any codec.
+    - *`"codec name"`*<sup>O</sup> (requires to be put in `"supported_codecs"` firstly): (e.g. `"av1"`) bitrate for this codec. It overrides `"any_codec"`.
+  - `"video_bitrate_matrix"`<sup>O</sup> (requires `"supported_resolutions"` with framerates): maximum bitrate based on resolutions and framerates from `"supported_resolutions"`.
+    - *`"CX x CY @ FPS"`*<sup>O</sup> (**one per resolution**): (e.g. `"1280x720@60":`) maximum bitrate for supported resolution of this service. Can be set per codec.
+      - `"any_codec"`<sup>O</sup>: bitrate for any codec.
+      - *`"codec name"`*<sup>O</sup> (requires to be put in `"supported_codecs"` firstly): (e.g. `"av1":`) bitrate for this codec. It overrides `"any_codec"`.
+  - `"audio_bitrate"`<sup>O</sup>: maximum audio bitrate. Can be set per codec.
+    - `"any_codec"`<sup>O</sup>: bitrate for any codecs.
+    - *`"codec name"`*<sup>O</sup> (requires to be put in `"supported_codecs"` firstly): (e.g. `"av1":`) bitrate for this codec. It overrides `"any_codec"`.
+- `"recommended"`<sup>O</sup>: recommended settings that become the defaults when using this service.
+  - `"keyint"`<sup>O</sup>: is now per video codec.
+    - `"any_codec"`<sup>O</sup>: keyframe interval for any codec.
+    - *`"codec name"`*<sup>O</sup> (requires to be put in `"supported_codecs"` firstly): (e.g. `"av1"`) keyframe interval for this codec. It overrides `"any_codec"`.
+  - `"bframes"`<sup>O</sup>: is now per video codec.
+    - `"any_codec"`<sup>O</sup>: b-frames for any codec.
```

Comment:
I think `"all_codecs"` or `"*"` might be better?

Reply:
This symbol was completely out of my thoughts when I wrote this. And yes, it might be better.